src/worker.cpp

```diff
@@ -374,7 +374,15 @@ void Worker::getSystemJson(std::string& systemJson,
 
             if (itrToHW != hwVersionList.end())
             {
-                systemJson += imKwdValue + "_" + (*itrToHW).second + ".json";
+                if (!(*itrToHW).second.empty())
+                {
+                    systemJson += imKwdValue + "_" + (*itrToHW).second +
```

Comment:
Instead if IM value can we use the value from "systemType" map itself?

Reply:
yes. we can use it
I will update